File: src/main/java/com/sfh/pokeRogueBot/model/modifier/impl/EnemyAttackStatusEffectChanceModifierItem.java
package com.sfh.pokeRogueBot.model.modifier.impl;

import com.sfh.pokeRogueBot.model.enums.StatusEffect;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class EnemyAttackStatusEffectChanceModifierItem extends ModifierItem {

    public static final String TARGET = "EnemyAttackStatusEffectChanceModifierType";

    private int chancePercent;
    private StatusEffect effect;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/impl/EnemyAttackStatusEffectChanceModifierItem.kt
package com.sfh.pokeRogueBot.model.modifier.impl

import com.sfh.pokeRogueBot.model.enums.StatusEffect

class EnemyAttackStatusEffectChanceModifierItem : ModifierItem() {

    companion object {
        const val TARGET = "EnemyAttackStatusEffectChanceModifierType"
    }

    var chancePercent: Int = 0
    var effect: StatusEffect? = null
}